Assembly-CSharp/Global/Sound/VoicePlayer.cs

```diff
@@ -25,6 +26,82 @@ public VoicePlayer()
 		this.stateTransition.Add(new SdLibSoundProfileStateGraph.TransitionDelegate(base.StopSound), SoundProfileState.CrossfadeIn, SoundProfileState.Stopped);
 	}
 
+	public static void PlayBattleVoice(int va_id, string text)
+    {
+		string vaPath = String.Format("Voices/{0}/battle/{2}/va_{1}", Localization.GetSymbol(), va_id, FF9StateSystem.Battle.battleMapIndex);
+		if (!String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".akb", true, true)) || !String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".ogg", true, false)))
+		{
+			SoundLib.VALog(String.Format("field:battle/{0}, feild:{0}, msg:{1}, text:{2} path:{3}", FF9StateSystem.Battle.battleMapIndex, va_id, text, vaPath));
+		}
+		else
+		{
+			// otherwise check for a general one
+			vaPath = String.Format("Voices/{0}/battle/general/va_{1}", Localization.GetSymbol(), va_id);
+			if (!String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".akb", true, true)) || !String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".ogg", true, false)){
+				SoundLib.VALog(String.Format("field:battle/general, msg:{0}, text:{1} path:{2}", va_id, text, vaPath));
+			}
+		}
+		var currentVAFile = new SoundProfile
+		{
+			Code = va_id.ToString(),
+			Name = vaPath,
+			SoundIndex = va_id,
+			ResourceID = vaPath,
+			SoundProfileType = SoundProfileType.Voice,
+			SoundVolume = 1f,
+			Panning = 0f,
+			Pitch = 0.5f
+		};
+
+		SoundLoaderProxy.Instance.Load(currentVAFile,
+		(soundProfile, db) =>
+		{
+			if (soundProfile != null)
+			{
+				SoundLib.voicePlayer.CreateSound(soundProfile);
+				SoundLib.voicePlayer.StartSound(soundProfile);
+				if (db.ReadAll().ContainsKey(soundProfile.SoundIndex))
+					db.Update(soundProfile);
+				else
+					db.Create(soundProfile);
+			}
+		},
+		ETb.voiceDatabase);
+		SoundLib.VALog(String.Format("field:battle/{3}, feild:{3}, msg:{0}, text:{1} path:{2}", va_id, text, vaPath, FF9StateSystem.Battle.battleMapIndex));
+	}
+
+	public static void PlayBattleScriptVoice(int cmd_id, string text)
```

Comment:
This method has a lot in common with `public static void PlayBattleVoice(int va_id, string text)`. Refactor these two methods to remove duplication.

Reply:
They are different because they are different paths and it's clearer where they are being called from as to what they are calling, so it was done for clear of reading because it's static it's not like it being duplicated for no reason it's just a fixed block of memory.

Reply:
Maybe, my message wasn't clear. What I meant is to refactor the *body* of these methods, not really the names.